exp/zapslog/slog_old.go

```diff
@@ -0,0 +1,190 @@
+//go:build go1 && !go1.21
```

Comment:
Typically, for Go-version-specific files, we've named them `foo_pre_go$version` and `foo_go$version`.
For example, we have https://github.com/uber-go/zap/blob/fcfd368624a286486e62d46deba7e42c20164616/array_go118.go.
(There's no pre-Go version for that.)

So this should probably be called slog_pre_go121.go.